File: frontend/src/types/campaign.ts
export type CampaignType = 'email' | 'social_media' | 'blog' | 'ad_copy' | 'landing_page'

export type Platform = 'facebook' | 'instagram' | 'twitter' | 'linkedin' | 'tiktok' | 'google_ads' | 'email'

export interface CampaignGenerate {
  product_name: string
  product_description: string
  target_audience: string
  campaign_type: CampaignType
  platforms: Platform[]
  tone?: string
  key_benefits?: string[]
  call_to_action?: string
}

export interface CampaignFromURL {
  website_url: string
  campaign_type: CampaignType
  platforms: Platform[]
  generate_images?: boolean
}

export interface CampaignContent {
  platform: Platform
  headline: string
  body: string
  hashtags?: string[]
  call_to_action: string
  image_suggestions?: string[]
  generated_image_url?: string
}

export interface Campaign {
  id: string
  name: string
  campaign_type: CampaignType
  product_name: string
  target_audience: string
  content: CampaignContent[]
  website_url?: string
  website_images?: { url: string; alt: string }[]
  created_at: string
  updated_at?: string
}

File: backend/app/models/campaign.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from enum import Enum


class CampaignType(str, Enum):
    EMAIL = "email"
    SOCIAL_MEDIA = "social_media"
    BLOG = "blog"
    AD_COPY = "ad_copy"
    LANDING_PAGE = "landing_page"


class Platform(str, Enum):
    FACEBOOK = "facebook"
    INSTAGRAM = "instagram"
    TWITTER = "twitter"
    LINKEDIN = "linkedin"
    TIKTOK = "tiktok"
    GOOGLE_ADS = "google_ads"
    EMAIL = "email"


class CampaignGenerate(BaseModel):
    """Request model for generating a campaign"""
    product_name: str
    product_description: str
    target_audience: str
    campaign_type: CampaignType
    platforms: List[Platform]
    tone: str = "professional"
    key_benefits: Optional[List[str]] = None
    call_to_action: Optional[str] = None


class CampaignContent(BaseModel):
    """Generated content for a specific platform"""
    platform: Platform
    headline: str
    body: str
    hashtags: Optional[List[str]] = None
    call_to_action: str
    image_suggestions: Optional[List[str]] = None


class CampaignCreate(BaseModel):
    """Model for creating a campaign manually"""
    name: str
    campaign_type: CampaignType
    content: List[CampaignContent]


class Campaign(BaseModel):
    """Full campaign model"""
    id: str
    name: str
    campaign_type: CampaignType
    product_name: str
    target_audience: str
    content: List[CampaignContent]
    created_at: datetime
    updated_at: Optional[datetime] = None
